File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        Map<Character, Integer> map = new HashMap<Character, Integer>();
        for (int i = 0; i < s.length(); i++) {
            char tmp = s.charAt(i);
            if (!map.containsKey(tmp)) {
                map.put(tmp, i);
                answer[i] = -1;
            } else {
                answer[i] = i - map.get(tmp);
                map.put(tmp, Math.max(i, map.get(tmp)));
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    ch_dict = {}
    for idx, ch in enumerate(s):
        if ch not in ch_dict.keys():
            answer.append(-1)
            ch_dict[ch] = idx
        else:
            answer.append(idx - ch_dict[ch])
            ch_dict[ch] = idx
    return answer